font/src/rusttype/mod.rs

```diff
@@ -72,6 +72,33 @@ impl ::Rasterize for RustTypeRasterizer {
     }
 
     fn get_glyph(&mut self, glyph_key: GlyphKey) -> Result<RasterizedGlyph, Error> {
+        match glyph_key.c {
+            super::UNDERLINE_CURSOR_CHAR => {
+                let metrics = self.metrics(glyph_key.font_key, glyph_key.size)?;
+                return super::get_underline_cursor_glyph(metrics.descent as i32, metrics.average_advance as i32);
+
+            }
+            super::BEAM_CURSOR_CHAR => {
+                let metrics = self.metrics(glyph_key.font_key, glyph_key.size)?;
+
+                return super::get_beam_cursor_glyph(
+                    (metrics.line_height + f64::from(metrics.descent)).round() as i32,
+                    metrics.line_height.round() as i32,
+                    metrics.average_advance.round() as i32
+                );
+            }
+            super::BEAM_CURSOR_CHAR | super::BOX_CURSOR_CHAR => {
```

Comment:
The Beam cursor is already covered in the previous branch, so it can be removed here.

```suggestion
            super::BOX_CURSOR_CHAR => {
```